### PR TITLE
Feature/logging

### PR DESCRIPTION
**Pull request by lumberj**
Motivation:
To provide structured, machine-parseable logging for production
deployments.

The bunyan logger will by default log in JSON in one line:

```
 {"name":"koa","hostname":"alan-lm","pid":39269,"req_id":"bbf85791-9169-43bd-a679-4214c8b9cea7","level":30,"req":{"method":"GET","url":"/accounts/test","headers":{"host":"localhost:3001","accept":"application/json","connection":"close"},"remoteAddress":"::ffff:127.0.0.1","remotePort":60024},"res":{"statusCode":200,"header":"HTTP/1.1 200 OK\r\nAccess-Control-Allow-Origin: *\r\nAccess-Control-Expose-Headers: link\r\nAccess-Control-Allow-Methods: GET,HEAD,PUT,POST,DELETE\r\nContent-Type: application/json; charset=utf-8\r\nContent-Length: 162\r\nDate: Tue, 12 Jul 2016 23:49:22 GMT\r\nConnection: close\r\n\r\n"},"duration":9,"msg":"  --> GET /accounts/test 200 9ms","time":"2016-07-12T23:49:22.811Z","v":0}
 ```

This can be parsed for human-readability using a JSON parser such as [`jq`](https://stedolan.github.io/jq/) or the [bunyan CLI](https://github.com/trentm/node-bunyan#cli-usage)

In this commit, `npm start` will by default pipe the JSON output to the
`bunyan` CLI. Example logging looks like:

```
 [2016-07-12T23:49:22.978Z]  INFO: koa/39269 on alan-lm:   <-- GET /connectors (req_id=0053ab9c-de7f-4782-a55f-a33ca6042f58, req.remoteAddress=::ffff:127.0.0.1, req.remotePort=60037)
    GET /connectors HTTP/1.1
    host: localhost:3001
    accept: application/json
    connection: close
[2016-07-12T23:49:22.982Z]  INFO: koa/39269 on alan-lm:   --> GET /connectors 200 4ms (req_id=0053ab9c-de7f-4782-a55f-a33ca6042f58, duration=4, req.remoteAddress=::ffff:127.0.0.1, req.remotePort=60037)
    GET /connectors HTTP/1.1
    host: localhost:3001
    accept: application/json
    connection: close
    --
    HTTP/1.1 200 OK
    HTTP/1.1 200 OK
...
```

Additionally, the request logger will log the `X-Request-ID` header and
store the result in the Koa context as `reqId` (e.g., `this.reqId`).
This is helpful for tracing requests across components (e.g., ledger ->
connector, connector -> ledger, client -> connector -> ledger)
See:
https://github.com/koajs/bunyan-logger#koabunyanloggerrequestidcontextopts

### CODE REVIEW

**Comment by MatthewPhinney** [comment]
LGTM

File: src/lib/app.js [hunk 1]
```diff
@@ -25,16 +25,25 @@ require('../services/auth')
...
     this.timerWorker = modules.timerWorker
     this.notificationBroadcaster = modules.notificationBroadcaster
 
     const koaApp = this.koa = websockify(koa())
     const router = this._makeRouter()
-    koaApp.use(logger())
-    koaApp.use(errorHandler({log: modules.log('error-handler')}))
+    koaApp.use(logger(this.log))
+    koaApp.use(logger.requestIdContext())
+    koaApp.use(logger.requestLogger({
```
Comment (lumberj) [inline src/lib/app.js]: I'm going to cleanup this middleware logger part

File: src/lib/config.js [hunk 2]
```diff
@@ -90,6 +90,15 @@ function parseKeysConfig () {
   }
 }
 
+function parseLogLevel () {
```
Comment (clark800) [inline src/lib/config.js]: Is this doing any parsing? Could it be `getLogLevel`?

**Comment by clark800** [comment]
LGTM

**Comment by lumberj** [comment]
Currently debugging an issue here with module scoping. `config` cannot be `require`d in the log module, so it ends up as an empty object, and `logLevel` cannot be accessed.

**Comment by clark800** [comment]
Looks like that is due to a circular dependency

**Comment by lumberj** [comment]
> Looks like that is due to a circular dependency

**Comment by lumberj** [comment]
Currently not working when tests fail. Looking into that.

**Pull request closed without merging**